core/src/Revolution/Filters/modOutputFilter.php

```diff
@@ -63,7 +63,7 @@ public function filter(&$element)
 
                 $this->log('Processing Modifier: ' . $m_cmd . ' (parameters: ' . $m_val . ')');
 
-                $output = trim($output);
+                $output = trim((string) $output);
```

Comment:
I'd rather see the solution test for string rather than cast all values to that type. If I'm not mistaken, there would be legitimate cases where the value being worked with is not a string (_e.g._, int or bool, etc.), so maybe this:
```suggestion
                $output = is_string($output) ? trim($output) : $output;
```

Reply:
It could be argued that all output filters should be operating only on strings. Perhaps it is worth looking at bypassing filtering altogether if the value is not a string?

Reply:
I don't _think_ so; what about the comparison filters (gt, lt, etc.) acting on the integer output of a snippet? The number TV, as it stands, does only output a string representation. But with that TV type  I could see there being an input option to force an integer, or a new filter that acts on a strict comparison. I think it just keeps your options (so to speak) open if you _don't_ coerce values passed to the filter to string. That's my 2¢ at least ;-)

Reply:
If I rephrase this to say scalar value, would you agree? Though technically, all elements in MODX that appear in content HAVE to return a string to be useful. You can obviously create snippets that return other (non-scalar) types in situations where you are using that snippet programmatically in PHP code, but when used in content, it is essentially useless. Since output filters are meant to act on the value returned to be used in content, I think my point is valid.

Reply:
Yeah, definitely agree with bypassing non-scalar vals. So maybe at the top of the filter method, add this:
``` php
if (!is_scalar($element->_output) {
    return;
}
```
Then still test for string to trim instead of coercing as I suggested above?